### PR TITLE
Fix ehdr/phdr endianness not being switched properly

### LINKED ISSUE #156: Exception with ELF MIPSII
Hey, when I set a breakpoint in an ELF MIPSII using remote debugging, pwndbg throws an exception (binary is attached):

[bender.gz](https://github.com/pwndbg/pwndbg/files/742558/bender.gz)

pwndbg> file bender_safe 
Reading symbols from bender_safe...(no debugging symbols found)...done.
pwndbg> b validate 
Breakpoint 1 at 0x401c84
pwndbg> c
Continuing.
Traceback (most recent call last):
  File "/home/user/pwndbg/pwndbg/events.py", line 111, in caller
    func()
  File "/home/user/pwndbg/pwndbg/vmmap.py", line 107, in explore_registers
    find(pwndbg.regs[regname])
...
gdb.MemoryError: Cannot access memory at address 0x34400000
Python Exception <class 'gdb.MemoryError'> Cannot access memory at address 0x34400000: 
Traceback (most recent call last):

### PR DESCRIPTION
**Pull request by disconnect3d**
Also fixes #156.

An explanation of the issue can be found in #202 and #239, as well as discussion/thoughts.

Here it is in action:
https://asciinema.org/a/35li7yv0qleb7xdj57svoelzm

### CODE REVIEW

File: pwndbg/arch.py [hunk 1]
```diff
@@ -30,6 +30,7 @@ def fix_arch(arch):
     return next(a for a in arches if a in arch)
 
 @pwndbg.events.start
+@pwndbg.events.new_objfile
```
Comment (disconnect3d) [inline pwndbg/arch.py]: We weren't reloading `pwndbg.arch.endian` when loading new object file (e.g. when connecting to remote target).

Comment (disconnect3d) [reply pwndbg/arch.py]: Hm.. isn't `@pwndbg.events.stop` redudant here?

File: pwndbg/elf.py [hunk 2]
```diff
@@ -30,16 +26,21 @@
 import pwndbg.proc
 import pwndbg.stack
 
+from imp import reload
```
Comment (disconnect3d) [inline pwndbg/elf.py]: Not sure if that is the proper way for cross Python compatibility. [An online cheatsheet](http://python-future.org/compatible_idioms.html#reload) suggests it.

However, in our own reload command, we use [reload from either `builtins` or from `imp`](https://github.com/pwndbg/pwndbg/blob/dev/pwndbg/commands/reload.py#L20-L23). Is there any reason to do so?

**Comment by zachriggle** [comment]
Import reload from six.moves

**Comment by disconnect3d** [comment]
I've imported `reload_module` from `six.moves` and removed `@pwndbg.events.stop` from `pwndbg.arch.update`.

**Comment by disconnect3d** [comment]
@zachriggle any comments or merge? :P

**Comment by zachriggle** [comment]
Ping me in 12 hours. I had planned to fix this myself but clearly I'm
slacking and should just merge this.

**Comment by disconnect3d** [comment]
@zachriggle ping

**Review by zachriggle — APPROVED** [review]

**Pull request merged**